src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala

```diff
@@ -19,9 +23,256 @@ class AppDeployIntegrationTest
+
+    When("The app is deployed")
+    val result = marathon.createApp(app)
+
+    Then("The app is created")
+    result.code should be (201) //Created
+    extractDeploymentIds(result) should have size 1
+    waitForEvent("deployment_success")
+  }
+
+  test("create a simple app with command health checks") {
+    Given("a new app")
+    val app = appProxy(testBasePath / "command-app", "v1", instances = 1, withHealth = false).
+      copy(healthChecks = Set(HealthCheck(protocol = Protocol.COMMAND, command = Some(Command("true")), gracePeriod = 20.second, interval = 1.second, maxConsecutiveFailures = 10)))
+
+    When("The app is deployed")
+    val result = marathon.createApp(app)
+
+    Then("The app is created")
+    result.code should be (201) //Created
+    extractDeploymentIds(result) should have size 1
+    waitForEvent("deployment_success")
+  }
+
+  test("list running apps and tasks") {
+    Given("a new app is deployed")
+    val appId = testBasePath / "app"
+    val app = appProxy(appId, "v1", instances = 2, withHealth = false)
+    marathon.createApp(app).code should be (201) //Created
+
+    When("the deployment has finished")
+    waitForEvent("deployment_success")
+
+    Then("the list of running apps can be fetched")
+    val apps = marathon.listAppsInBaseGroup
+    apps.code should be(200)
+    apps.value should have size 1
+
+    val tasks = marathon.tasks(appId)
+    tasks.code should be(200)
+    tasks.value should have size 2
+  }
+
+  test("create an app that fails to deploy") {
+    Given("a new app that is not healthy")
+    val appId = testBasePath / "failing"
+    val check = appProxyCheck(appId, "v1", state = false)
+    val app = appProxy(appId, "v1", instances = 1, withHealth = false).
+      copy(healthChecks = Set(HealthCheck(gracePeriod = 20.second, interval = 1.second, maxConsecutiveFailures = 10)))
+
+    When("The app is deployed")
+    val create = marathon.createApp(app)
+
+    Then("The deployment can not be finished")
+    create.code should be (201) //Created
+    extractDeploymentIds(create) should have size 1
+    intercept[AssertionError] {
+      waitForEvent("deployment_success")
+    }
+
+    When("The app is deleted")
+    val delete = marathon.deleteApp(appId, force = true)
+    delete.code should be (200)
+    waitForChange(delete)
+    marathon.listAppsInBaseGroup.value should have size 0
+  }
+
+  test("update an app") {
+    Given("a new app")
+    val appId = testBasePath / "app"
+    val v1 = appProxy(appId, "v1", instances = 1, withHealth = false)
+    marathon.createApp(v1).code should be (201)
+    waitForEvent("deployment_success")
+    val before = marathon.tasks(appId)
+
+    When("The app is updated")
+    val update = marathon.updateApp(v1.id, AppUpdate(cmd = appProxy(appId, "v2", 1).cmd))
+
+    Then("The app gets updated")
+    update.code should be (200)
+    waitForEvent("deployment_success")
+    val after = marathon.tasks(appId)
+    waitForTasks(appId, before.value.size)
```

Comment:
You do not check that, do you?

You could check it by inspecting `after.value.map(_.version)` for example.